### ISSUE TITLE
🐛 BUG: wrangler pages dev runtime failure

### ISSUE THREAD

**Issue by Sammy-T** [cloudflare/workers-sdk#6169]
### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

Wrangler v3.62.0

### What version of Node are you using?

18.19.1

### What operating system and version are you using?

Windows 10

### Observed behavior
`wrangler pages dev [directory]` fails on v3.62.0. The exact same command is successful using v3.61.0 though.

### Expected behavior
Successfully run the local server.

### Steps to reproduce
- Initialize node project directory.
- Create test page in subdirectory. ex. `site/index.html`
- Create package.json script. `"dev": "wrangler pages dev site"`
- Install wrangler. ex. `pnpm i -D wrangler` or `npm i -D wrangler`
- Run package.json script. `pnpm run dev` or `npm run dev` (neither worked for me.)

### Please provide a link to a minimal reproduction

<https://github.com/Sammy-T/wrangler-pages-err>

### Please provide any relevant error logs

```
 ⛅️ wrangler 3.62.0
-------------------

▲ [WARNING] No compatibility_date was specified. Using today's date: 2024-06-29.

  ❯❯ Add one to your wrangler.toml file: compatibility_date = "2024-06-29", or
  ❯❯ Pass it in your terminal: wrangler pages dev [<DIRECTORY>] --compatibility-date=2024-06-29

  See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information.


No functions. Shimming...
⎔ Starting local server...

C:\Users\Sammy\Desktop\New folder\node_modules\.pnpm\wrangler@3.62.0\node_modules\wrangler\wrangler-dist\cli.js:29747
            throw a;
            ^

MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
    at #assembleAndUpdateConfig (C:\Users\Sammy\Desktop\New folder\node_modules\.pnpm\miniflare@3.20240620.0\node_modules\miniflare\dist\src\index.js:9161:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Mutex.runWith (C:\Users\Sammy\Desktop\New folder\node_modules\.pnpm\miniflare@3.20240620.0\node_modules\miniflare\dist\src\index.js:3521:16) {
  code: 'ERR_RUNTIME_FAILURE',
  cause: undefined
}

Node.js v18.19.1
 ELIFECYCLE  Command failed with exit code 7.
```

**Comment by petebacondarwin** [comment]
I just tried to reproduce this on MacOS and Windows and it worked both times. For example on Windows:

```
PS C:\Users\Admin\dev\temp\foo> pnpm run dev

> foo@1.0.0 dev C:\Users\Admin\dev\temp\foo
> wrangler pages dev site


 ⛅️ wrangler 3.62.0
-------------------

▲ [WARNING] No compatibility_date was specified. Using today's date: 2024-06-29.

  ❯❯ Add one to your wrangler.toml file: compatibility_date = "2024-06-29", or
  ❯❯ Pass it in your terminal: wrangler pages dev [<DIRECTORY>] --compatibility-date=2024-06-29    
  
  See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information.


No functions. Shimming...
⎔ Starting local server...
[wrangler:inf] Ready on http://127.0.0.1:8788
[wrangler:inf] GET / 200 OK (19ms)
[wrangler:inf] GET /favicon.ico 200 OK (19ms)
```

**Comment by petebacondarwin** [comment]
Could you create a reproduction and push it to a github repository so that we can try it out?

**Comment by Sammy-T** [comment]
Sure, I updated the description with a link to it.

**Comment by penalosa** [comment]
@Sammy-T I'm unable to reproduce this with your reproduction repo. Could you try upgrading to the latest version of Node & Wrangler? What's the specific version number of your Windows 10 install?

**Comment by Sammy-T** [comment]
I updated both wrangler and node and it still fails. Here's my OS info and the received error.

Windows 10 22H2 Version 10.0.19045 Build 19045

```
⛅️ wrangler 3.63.1
-------------------

▲ [WARNING] No compatibility_date was specified. Using today's date: 2024-07-08.

  ❯❯ Add one to your wrangler.toml file: compatibility_date = "2024-07-08", or
  ❯❯ Pass it in your terminal: wrangler pages dev [<DIRECTORY>] --compatibility-date=2024-07-08

  See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information.


No Functions. Shimming...
⎔ Starting local server...
╭─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ [b] open a browser, [d] open Devtools, [c] clear console, [x] to exit                                               │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
C:\Users\Sammy\Documents\GitHub\temp\wrangler-pages-err\node_modules\wrangler\wrangler-dist\cli.js:29765
            throw a;
            ^

MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
    at #assembleAndUpdateConfig (C:\Users\Sammy\Documents\GitHub\temp\wrangler-pages-err\node_modules\miniflare\dist\src\index.js:9178:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Mutex.runWith (C:\Users\Sammy\Documents\GitHub\temp\wrangler-pages-err\node_modules\miniflare\dist\src\index.js:3521:16) {
  code: 'ERR_RUNTIME_FAILURE',
  cause: undefined
}

Node.js v20.15.1
```

**Comment by cpakken** [comment]
Same for me wrangler@3.61.0 works but anything later breaks with the same error

**Comment by Sammy-T** [comment]
> Same for me wrangler@3.61.0 works but anything later breaks with the same error (windows 10 22h2) EDIT: Solved it thanks to [#6170 (comment)](https://github.com/cloudflare/workers-sdk/issues/6170#issuecomment-2245209918)
> 
> https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170 Update latest Visual C++ Redistributable and it works with wrangler@3.66.0

That seems to have fixed it for me. Thanks, @cpakken.
